### PR TITLE
Feat/water 2768 de minimis

### PR DESCRIPTION
**Pull request by james-siteclick**
Reads de-minimis flag from charge module customer financial year summary and maps to service models at Invoice level.
Reads de-minimis flag from charge module transactions and maps to service models at Transactions level.  Stores flag in water.billing_transactions.is_deminimis.
When running supplementary billing, does not credit transactions that were subject to de-minimis

### CODE REVIEW

File: test/modules/billing/mappers/totals.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ experiment('modules/billing/mappers/totals', () => {
 
     experiment('when the customer is present in the bill run summary', () => {
       beforeEach(async () => {
-        result = totalsMapper.chargeModuleBillRunToInvoiceModel(data.billRun, INVOICE_ACCOUNT_NUMBER, 2021);
+        result = totalsMapper.chargeModuleBillRunToInvoiceModel(data, INVOICE_ACCOUNT_NUMBER, 2021);
```
Comment (timtamimi) [inline test/modules/billing/mappers/totals.js:92]: I doubt it's a big deal, but I feel like `2021` here should be an env variable

File: src/modules/billing/services/batch-service.js [hunk 2]
```diff
@@ -150,28 +155,56 @@ const saveInvoicesToDB = async batch => {
...
+ * @param {Batch} batch
+ * @return {Promise}
+ */
+const persistTotals = batch => {
+  const changes = batch.totals.pick([
+    'invoiceCount',
+    'creditNoteCount',
+    'netTotal'
+  ]);
+  return newRepos.billingBatches.update(batch.id, changes, false);
```
Comment (DGajewska) [inline src/modules/billing/services/batch-service.js:184]: I think you've got an extra param

File: src/modules/billing/services/transactions-service.js [hunk 3]
```diff
@@ -79,19 +77,56 @@ const setErrorStatus = transactionId =>
...
+ * @return {Promise}
+ */
+const clearDeMinimisByTransactionIds = partialRight(updateDeMinimis, false);
+
+/**
+ * Sets the de-minimis flags by transaction IDs
+ * @param {Array<String>} externalIds
+ * @return {Promise}
+ */
+const setDeMinimisByTransactionIds = partialRight(updateDeMinimis, true);
```
Comment (DGajewska) [inline src/modules/billing/services/transactions-service.js:95]: I don't think these functions are used anywhere. Did you mean to use them below?

File: src/modules/billing/services/batch-service.js (hunk 2 above)
Comment (james-siteclick) [reply src/modules/billing/services/batch-service.js]: Thanks, this should be fixed now

File: src/modules/billing/services/transactions-service.js (hunk 3 above)
Comment (james-siteclick) [reply src/modules/billing/services/transactions-service.js]: Thanks, this should be fixed now

**Pull request merged**